lib/elixir/test/elixir/module/types/infer_test.exs

```diff
@@ -14,10 +14,18 @@ defmodule Module.Types.InferTest do
 
   defmacrop quoted_guard(guards, context) do
     quote do
-      of_guard(unquote(Macro.escape(guards)), new_stack(), unquote(context))
+      of_guard(unquote(Macro.escape(expand_guards(guards))), new_stack(), unquote(context))
     end
   end
 
+  defp expand_guards(guards) do
+    x = nil
+    _ = x
```

Comment:
Is there a better to introduce variables to the environment? Even if I manually add them to `__ENV__.vars` I still get errors.

Reply:
@ericmj I think the most reliable way would be for you to expand a whole expression, such as `fn x when unquote(guards) -> :ok end` and then you extract the guards out. WDTY?